api/client/inspect.go

```diff
@@ -18,42 +18,72 @@ import (
 func (cli *DockerCli) CmdInspect(args ...string) error {
 	cmd := cli.Subcmd("inspect", "CONTAINER|IMAGE [CONTAINER|IMAGE...]", "Return low-level information on a container or image", true)
 	tmplStr := cmd.String([]string{"f", "#format", "-format"}, "", "Format the output using the given go template")
+	inspectType := cmd.String([]string{"-type"}, "", "Return JSON for specified type, (e.g image or container)")
+
 	cmd.Require(flag.Min, 1)
 
 	cmd.ParseFlags(args, true)
 
 	var tmpl *template.Template
+	var err error
+	var obj []byte
+
 	if *tmplStr != "" {
-		var err error
 		if tmpl, err = template.New("").Funcs(funcMap).Parse(*tmplStr); err != nil {
 			return StatusError{StatusCode: 64,
 				Status: "Template parsing error: " + err.Error()}
 		}
 	}
 
+	if *inspectType != "" && *inspectType != "container" && *inspectType != "image" {
+		return fmt.Errorf("%s is not a valid value for --type", *inspectType)
+	}
+
 	indented := new(bytes.Buffer)
 	indented.WriteString("[\n")
 	status := 0
 	isImage := false
+	isContainer := false
 
 	for _, name := range cmd.Args() {
-		obj, _, err := readBody(cli.call("GET", "/containers/"+name+"/json", nil, nil))
-		if err != nil {
+
+		isContainer = false
+
+		if *inspectType == "" || *inspectType == "container" {
+			obj, _, err = readBody(cli.call("GET", "/containers/"+name+"/json", nil, nil))
+			if err == nil {
+				isContainer = true
+			} else {
+				if *inspectType == "container" {
+					if strings.Contains(err.Error(), "No such") {
+						fmt.Fprintf(cli.err, "Error: No such container: %s\n", name)
+					} else {
+						fmt.Fprintf(cli.err, "%s", err)
+					}
+					status = 1
+					continue
+				}
+			}
+
+		}
+
+		if !isContainer {
```

Comment:
@duglin I cannot change that to, `if *inspectType == "" || *inspectType == "image" {`
because then in the default behavior it will go for both container and image JSON {will show the image JSON only in the output), which will change the existing behavior.

There is no `obj != nil` check in the existing upstream code. Would you like me to add this additional check ?